sisl/geom/nanoribbon.py

```diff
@@ -154,3 +159,461 @@ def zgnr(width, bond=1.42, atoms=None):
     agnr : armchair graphene nanoribbon
     """
     return graphene_nanoribbon(width, bond, atoms, kind='zigzag')
+
+
+Heteroribbon_section = namedtuple("Section", ["W", "L", "shift", "atoms"], defaults=(None, 1, 0, None))
+
+
+@set_module("sisl.geom")
+def heteroribbon(sections, bond, atoms=None, kind="armchair"):
```

Comment:
`atoms` should not be optional, what happens for sections that do not have one?
Hmm...